apollo-android-support/src/main/java/com/apollographql/apollo/cache/normalized/sql/SqlNormalizedCache.java

```diff
@@ -102,40 +99,35 @@
     return changedKeys;
   }
 
-  @SuppressWarnings("ResultOfMethodCallIgnored")
   @Override public void clearAll() {
-    //noinspection ResultOfMethodCallIgnored
-    nextCache().apply(new Action<NormalizedCache>() {
-      @Override public void apply(@NotNull NormalizedCache cache) {
-        cache.clearAll();
-      }
-    });
+    NormalizedCache nextCache = getNextCache();
+    if (nextCache != null) {
+      nextCache.clearAll();
+    }
     clearCurrentCache();
   }
 
   @Override public boolean remove(@NotNull final CacheKey cacheKey, final boolean cascade) {
```

Comment:
I have to double check entire logic of this function, after quick look at the logic that was before looks suspicious.
It looks like we remove only from the `nextCache` if it present and never from current instance 🤔

Reply:
Do you mean the current change is suspicious or the old behavior is? This code is quite well tested.

Reply:
From what I understand, `return result || ...` was doing the same. If the result is true, the rest will never be executed. 

On a second look, I don't think the tests cover having a nextCache scenario

Reply:
I did another iteration to keep it much closer to the original implementation. Behavior wise, it should be the same but I think this will make it easier to review. Let me know if you see any issues.

I am personally not 100% sure how should the implementation be with the `nextCache` involved.

Reply:
No sorry for the confusion, I had to mention that it's not about your changes, I meant the logic that we had before.

The issue is that we are removing the next chained cache (`nextCache()`) and if it  succeed we skip the current one as `return result || deleteRecord` won't call `deleteRecord`.

The reason that it's not an issue is because usually `SqlNormalizedCache` is the last one in the chain but implementation is definitely not correct.

I will fix it in separate PR, let's keep it for now as is.